homeassistant/components/electric_kiwi/select.py

```diff
@@ -0,0 +1,99 @@
+"""Support for Electric Kiwi hour of free power."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Final
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTRIBUTION, DOMAIN
+from .coordinator import ElectricKiwiHOPDataCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+ATTR_EK_HOP_SELECT = "hop_select"
+
+
+@dataclass
+class ElectricKiwiHOPSelectDescriptionMixin:
+    """Define an entity description mixin for select entities."""
+
+    options_dict: dict[str, int] | None
+
+
+@dataclass
+class ElectricKiwiHOPSelectDescription(
+    SelectEntityDescription, ElectricKiwiHOPSelectDescriptionMixin
+):
+    """Class to describe an Electric Kiwi select entity."""
+
+
+HOP_SELECT_TYPE: Final[tuple[ElectricKiwiHOPSelectDescription, ...]] = (
```

Comment:
If this is the only HOP select type, you can take it out of the tuple

Reply:
Yeah it will be the only HOP one, there will be others but they will be different types, will tidy it up.